### PR TITLE
Add MCP server views to Poke

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Debugging which MCP servers views are created locally gets quite hard since the data is scattered across different tables.
- This PR adds a DataTable in Poke to display MCP server views, quite similarly to data source views, in the workspace page and in the space page.

## Tests

- Tested locally.

## Risk

- Potentially breaks Poke, there are many edge cases that my local setup does not cover.
- Will check thoroughly in production.

## Deploy Plan

- Deploy front.

### CODE REVIEW

File: front/components/poke/spaces/columns.tsx [hunk 1]
```diff
@@ -78,7 +78,7 @@ export function makeColumnsForSpaces(
       },
       header: ({ column }) => {
         return (
-          <div className="flex space-x-2">
+          <div className="flex items-center space-x-2">
```
Comment (aubin-tchoi) [inline front/components/poke/spaces/columns.tsx:81]: couldn't really find how to fix this in a more generic manner 😢

File: front/components/poke/mcp_server_views/view.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                <PokeTableHead>Server Description</PokeTableHead>
+                <PokeTableCell>
+                  {mcpServerView.server.description}
+                </PokeTableCell>
+              </PokeTableRow>
+              <PokeTableRow>
+                <PokeTableHead>Space</PokeTableHead>
+                <PokeTableCellWithLink
+                  href={`/poke/${owner.sId}/spaces/${mcpServerView.spaceId}`}
+                  content={`${"space" in mcpServerView ? mcpServerView.space?.name || "" : ""} (${mcpServerView.spaceId})`}
```
Comment (flvndvd) [inline front/components/poke/mcp_server_views/view.tsx]: Don't we always have a space?

File: front/components/poke/spaces/columns.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/poke/spaces/columns.tsx]: Wrapper?

File: front/types/poke/index.ts [hunk 3]
```diff
@@ -32,6 +34,13 @@ export type PokeDataSourceViewType = DataSourceViewType &
     space: PokeSpaceType;
   };
 
+export type PokeMCPServerViewType = MCPServerViewType &
+  // Consequence of small inconsistencies in typing between MCPServerViewType and
+  // DataSourceViewType, hard to fix and only affects this type for now.
+  Omit<PokeItemBase, "id"> & {
```
Comment (flvndvd) [inline front/types/poke/index.ts:40]: Ack, would be nice to align the with all the other resources though.

**Review by flvndvd — APPROVED** [review]

File: front/types/poke/index.ts (hunk 3 above)
Comment (aubin-tchoi) [reply front/types/poke/index.ts:40]: opened a thread 🍿
